recipes/flann/build.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# cannot build flann from within the source directory
+mkdir build
+cd build
+
+# On OSX, we need to ensure we're using conda's gcc/g++
+if [[ `uname` == Darwin ]]; then
+    export CC=gcc
+    export LD=gcc
+    export CXX=g++
+fi
+
+cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX -DBUILD_MATLAB_BINDINGS:BOOL=OFF -DBUILD_PYTHON_BINDINGS:BOOL=OFF -DBUILD_EXAMPLES:BOOL=OFF -DBUILD_DOC:BOOL=OFF
+
```

Comment:
This ended up being too much of a pain as the tests required NumPy (thus pulling in Python) defeating the OP's intent. Also, there was a known test failure that occurred, as well. In addition, the tests were pulling down data to test on. However, the code doing this (not distributed) was written in Python and was Python 2.x compatible only. To get all of this to work we would have to patch out all these cases and fix their scripts, which begins to question the value of running the tests at all.